placement/algo/sharded_helper.go

```diff
@@ -738,3 +738,80 @@ func newShards(shardIDs []uint32) []shard.Shard {
 	}
 	return r
 }
+
+func removeInstanceFromList(list []placement.Instance, instanceID string) []placement.Instance {
+	for i, instance := range list {
+		if instance.ID() == instanceID {
+			last := len(list) - 1
+			list[i], list[last] = list[last], list[i]
+			return list[:last]
+		}
+	}
+	return list
+}
+
+func markShardAvailable(p placement.Placement, instanceID string, shardID uint32, nowNanos int64, shouldCheckCutoverTime bool) (placement.Placement, error) {
+	instance, exist := p.Instance(instanceID)
+	if !exist {
+		return nil, fmt.Errorf("instance %s does not exist in placement", instanceID)
+	}
+
+	shards := instance.Shards()
+	s, exist := shards.Shard(shardID)
+	if !exist {
+		return nil, fmt.Errorf("shard %d does not exist in instance %s", shardID, instanceID)
+	}
+
+	if s.State() != shard.Initializing {
+		return nil, fmt.Errorf("could not mark shard %d as available, it's not in Initializing state", s.ID())
+	}
+
+	if shouldCheckCutoverTime && s.CutoverNanos() > nowNanos {
+		return nil, fmt.Errorf("could not mark shard %d as available, it's cutover time %d is scheduled after now %d", s.ID(), s.CutoverNanos(), nowNanos)
+	}
+
+	sourceID := s.SourceID()
+	shards.Add(shard.NewShard(shardID).SetState(shard.Available))
+
+	// there could be no source for cases like initial placement
```

Comment:
nit: start comment with capital letter and end with period perhaps?

Reply:
will do